sammo.py

```diff
@@ -4,17 +4,44 @@
 __copyright__ = "Copyright (c) 2021 Hytech Imaging"
 
 from .src.gui.session import SammoActionSession
+from .src.gui.onOffEffort import SammoActionOnOffEffort
+from .src.core.session import SammoSession
+from qgis.PyQt.QtWidgets import QToolBar
+from qgis.core import QgsFeature
 
 
 class Sammo:
     def __init__(self, iface):
         self.iface = iface
-        self.toolBar = self.iface.addToolBar("Sammo ToolBar")
-        self.actionSession = SammoActionSession(self.iface.mainWindow(), self.toolBar)
+        self._toolBar: QToolBar = self.iface.addToolBar("Sammo ToolBar")
+        self._session = SammoSession()
+
+        self.actionSession = SammoActionSession(iface.mainWindow(), self._toolBar)
+        self.actionSession.createSignal.connect(self.onCreateSession)
+
+        self.actionOnOffSession = SammoActionOnOffEffort(iface.mainWindow(), self._toolBar)
+        self.actionOnOffSession.onStartEffortSignal.connect(self.onStartEffort)
+        self.actionOnOffSession.onStopEffortSignal.connect(self.onStopEffort)
+        self.actionOnOffSession.onAddFeatureToEnvironmentTableSignal.connect(self.onAddFeatureToEnvironmentTableSignal)
 
     def initGui(self):
-        self.actionSession.initGui()
+        pass
```

Comment:
We can remove `initGui`

Reply:
I tried but got a QGIS error. It seems to be a mandatory method

Reply:
Ok